pkg/proxy/switch.go

```diff
@@ -425,6 +417,20 @@ func (s *SwitchSession) Bridge(userConn UserConnection, srvConn srvconn.ServerCo
 		lastActiveTime = time.Now()
 	}
 }
+func (s *SwitchSession) disconnection(room *exchange.Room, parser *Parser, replayRecorder *ReplyRecorder, msg string) {
+	msg = utils.WrapperWarn(msg)
+	replayRecorder.Record([]byte(msg))
+
+	roomMessage := &exchange.RoomMessage{Event: exchange.DataEvent, Body: []byte("\n\r" + msg)}
+	if parser.zmodemParser.IsStartSession() {
+		expectedSize := len(zmodem.SkipSequence) + len(zmodem.CancelSequence)
+		roomMessage.Body = make([]byte, 0, expectedSize)
+		roomMessage.Body = append(roomMessage.Body, zmodem.SkipSequence...)
+		roomMessage.Body = append(roomMessage.Body, zmodem.CancelSequence...)
+	}
+
+	room.Broadcast(roomMessage)
+}
 
 func (s *SwitchSession) recordSessionFinished(reason model.SessionLifecycleReasonErr) {
 	logObj := model.SessionLifecycleLog{Reason: string(reason)}
```

Comment:
This is a draft code with some modifications suggested to improve its structure and style. The function "disconnection" seems unnecessary given that the intention of closing or ending sessions already occurs directly within this context.

Here are my observations:

- There's no need for an `else` branch if there's only one case; you can use `switch`.
- Use consistent formatting and indentation.
- Remove the unused package import `[1, ..., n]`.

The changes would be minimal because the code mostly looks fine on first sight. It does not suggest any major problems nor significant efficiency gains. However, I've mentioned some stylistic adjustments that aim to make the code cleaner but did not introduce new errors or bugs.

Please review it and decide whether these enhancements should happen as part of the official implementation or they might cause confusion due to their unexpected effects.

As we work towards implementing best coding practices, remember always to consider accessibility when writing documentation.